website/docs/automation/gitops/argocd-codecommit/cluster_bootstrap.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "Cluster bootstrap"
+sidebar_position: 15
+---
+
+An AWS CodeCommit repository has already been created for you and Argo CD has already been deployed to the EKS cluster so let's create necessary Argo CD components for our application.
+
+Create an Argo CD `Application`:
+
+```file
+automation/gitops/argocd-codecommit/argocd-application.yaml
+```
+
+Let's break down the manifest above:
+
+- First we tell ArgoCD what will be the `name` of our application
+- After that we define which Git repository `source` to use to get the desired state of our application. We define `repoUrl` of the AWS CodeCommit repository, a repository branch with `targetRevision`
+- Finally we define a `path` to application manifests within the repository
+
+Let's apply the manifest:
+
+```bash
+$ envsubst < /workspace/modules/automation/gitops/argocd-codecommit/argocd-application.yaml | kubectl apply -f -
```

Comment:
Lets change this to use kustomize to substitute in the environment variables to make it consistent with all our other labs. I would suggest moving the Argo application in to its own directory (like `environment/workspace/modules/automation/gitops/argocd-codecommit/application`) then apply it with `kubectl apply -k`.

You can see an example of us doing this with a security group policy here:

https://github.com/aws-samples/eks-workshop-v2/tree/main/environment/workspace/modules/networking/securitygroups-for-pods/sg

Please also add the Argo Application CRD to the end of this list so it gets cleaned up by our reset-environment script https://github.com/aws-samples/eks-workshop-v2/blob/main/environment/bin/reset-environment#L43

Reply:
Created separate folder for Argo CD kustomization and changed to solution with "replacements" which should be future proof.

Reply:
Addedd Argo CD CRD to reset-environment